modules/azure-gaming/game-dev-vmss/main.bicep

```diff
@@ -0,0 +1,248 @@
+@secure()
+param fileShareStorageAccountKey string = ''
+
+@description('File Share name')
+param fileShareName string = ''
+
+@description('Perforce Port address')
+param p4Port string = ''
+
+@description('Perforce User')
+param p4Username string = ''
+
+@description('Perforce User password')
+@secure()
+param p4Password string = ''
+
+@description('Perforce Client Workspace')
+param p4Workspace string = ''
+
+@description('Perforce Stream')
+param p4Stream string = ''
+
+@description('Perforce Depot Client View mappings')
+param p4ClientViews string = ''
+
+@description('Incredibuild License Key')
+@secure()
+param ibLicenseKey string = ''
+
+@description('GDK Version')
+@allowed([
+  'June_2022_Update_1'
+  'March_2022_Update_1'
+  'October_2021_Update_5'
+  'June_2021_Update_9'
+])
+param gdkVersion string = 'June_2022_Update_1'
+
+@description('Use VM to sysprep an image from')
+param useVmToSysprepCustomImage bool = false
+
+@description('Remote Access technology')
+@allowed([
+  'RDP'
+  'Teradici'
+  'Parsec'
+])
+param remoteAccessTechnology string = 'RDP'
+
+@description('Teradici Registration Key')
+@secure()
+param teradiciRegKey string = ''
+
+@description('Parsec Team ID')
+param parsecTeamId string = ''
+
+@description('Parsec Team Key')
+@secure()
+param parsecTeamKey string = ''
+
+@description('Parsec Hostname')
+param parsecHost string = ''
+
+@description('Parsec User Email')
+param parsecUserEmail string = ''
+
+@description('Parsec Is Guest Access')
+param parsecIsGuestAccess bool = false
+
+@description('Virtual Network Resource Name')
+param vnetName string = 'vnet-${vmssName}'
+
+@description('Virtual Network Subnet Name')
+param subnetName string = 'subnet-${vmssName}'
+
+@description('Virtual Network Security Group Name')
+param networkSecurityGroupName string = 'nsg-${vmssName}'
+
+@description('Virtual Network Address Prefix')
+param vnetAddressPrefix string = '172.17.72.0/24' //Change as needed
+
+@description('Virtual Network Subnet Address Prefix')
+param subnetAddressPrefix string = '172.17.72.0/25' // 172.17.72.[0-128] is part of this subnet
+
+var customData = format('''
+fileShareStorageAccount={0}
+fileShareStorageAccountKey={1}
+fileShareName={2}
+p4Port={3}
+p4Username={4}
+p4Password={5}
+p4Workspace={6}
+p4Stream={7}
+p4ClientViews={8}
+ibLicenseKey={9}
+gdkVersion={10}
+useVmToSysprepCustomImage={11}
+remoteAccessTechnology={12}
+teradiciRegKey={13}
+parsecTeamId={14}
+parsecTeamKey={15}
+parsecHost={16}
+parsecUserEmail={17}
+parsecIsGuestAccess={18}
+deployedFromSolutionTemplate={19}
+''', fileShareStorageAccount, fileShareStorageAccountKey, fileShareName, p4Port, p4Username, p4Password, p4Workspace, p4Stream, p4ClientViews, ibLicenseKey, gdkVersion, useVmToSysprepCustomImage, remoteAccessTechnology, teradiciRegKey, parsecTeamId, parsecTeamKey, parsecHost, parsecUserEmail, parsecIsGuestAccess, false)
+
+var vmssImgProduct = 'game-dev-vm'
+
+module vnet './nestedtemplates/virtualNetworks.bicep'  = {
```

Comment:
this is a nit pick, but I think we should get away from the language of `nestedTemplates`. Can we change the path to `modules`?